source/server/connection_handler_impl.cc

```diff
@@ -477,6 +478,35 @@ void ConnectionHandlerImpl::ActiveTcpListener::newConnection(
 
   auto transport_socket = filter_chain->transportSocketFactory().createTransportSocket(nullptr);
   stream_info->setDownstreamSslConnection(transport_socket->ssl());
+
+  if constexpr (Network::Win32SupportsOriginalDestination()) {
+    ASSERT(socket->addressProvider().localAddressRestored() &&
```

Comment:
Isn't `socket->addressProvider().localAddressRestored()` false for all non-original-dst listeners?

Reply:
You are correct, regardless I removed the assert in 583512a9750f17a04e7c1de4d0a428588fda08a1 since we will be guarding this in `listener_impl`.